docs/plugins/create-plugin.md

```diff
@@ -154,3 +154,322 @@ gem "fastlane-plugin-[plugin_name]", git: "https://github.com/[user]/[plugin_nam
 #### Multiple actions in one plugin
 
 Let's assume you work on a _fastlane_ plugin for project management software. You could call it `fastlane-plugin-pm` and it may contain any number of actions and helpers, just add them to your `actions` folder. Make sure to mention the available actions in your plugin's `README.md`.
+
+#### Using FastlaneCore::Configuration
+
+Note that it is not necessary to `require "fastlane_core"` in a plugin.
+
+Most actions accept one or more parameters to customize their behavior. Actions define their
+parameters in an `available_options` method. This method returns an array of `FastlaneCore::ConfigItem`
+objects to describe supported options. Each option is declared by creating a new
+`ConfigItem`, e.g.:
+
+```ruby
+FastlaneCore::ConfigItem.new(
+  key: :file,
+  env_name: "MY_NEW_ACTION_FILE",
+  description: "A file to operate on",
+  type: String,
+  optional: false
+)
+```
+
+This declares a `file` option for use with the action in a Fastfile, e.g.:
+
+```ruby
+my_new_action(file: "file.txt")
+```
+
+If the optional `env_name` is present, an environment variable with the specified
+name may also be used in place of an option in the Fastfile:
+
+```bash
+MY_NEW_ACTION_FILE=file.txt fastlane run my_new_action
+```
+
+The `type` argument to the `FastlaneCore::ConfigItem` initializer specifies the
+name of a Ruby class representing a standard
+data type. Supplied arguments will be coerced to the specified type. Some standard types
+support default conversions.
+
+##### Boolean parameters
+
+Ruby does not have a single class to represent a Boolean type. When specifying
+Boolean parameters, use `is_string: false`, without specifying a `type`, e.g.:
+
+```ruby
+FastlaneCore::ConfigItem.new(
+  key: :commit,
+  env_name: "MY_NEW_ACTION_COMMIT",
+  description: "Commit the results if true",
+  optional: true,
+  default_value: false,
+  is_string: false
+)
+```
+
+When passing a string value, e.g. from an environment variable, certain set
+string values are recognized:
+
+```bash
+MY_NEW_ACTION_COMMIT=true
+MY_NEW_ACTION_COMMIT=false
+MY_NEW_ACTION_COMMIT=yes
+MY_NEW_ACTION_COMMIT=no
+```
+
+These values may also be passed in all caps, e.g. `MY_NEW_ACTION_COMMIT=YES`.
+
+##### Array parameters
```

Comment:
This is all really awesome! However those are not plugin specific docs, right? But also when building your own actions, or when contributing to the built-in actions, right?

Reply:
I agree. This is more general than plugins.